### ISSUE TITLE
Subscribing to a non-existent topic doesn't error out

### ISSUE THREAD

**Issue by sershe84** [confluentinc/confluent-kafka-dotnet#1144]
Description
===========
When calling subscribe on a non-existent topic, there's no failure from either subscribe or consume; we just don't get any events.
Subscribe call returns without any error.
Side note, appending to a non-existent topic fails just fine on the same code (from the same process, actually)

How to reproduce
================
```
var cfg = new ConsumerConfig()
            {
                BootstrapServers = ...,
                GroupId = "DefaultW",
                ClientId = "myclient21334234",
                EnableAutoOffsetStore = false,
                EnableAutoCommit = false
            };

            var cb = new ConsumerBuilder<byte[], byte[]>(cfg)
                .SetErrorHandler((_, e) => logger.LogError($"Kafka error callback {e}"))
                .SetPartitionsRevokedHandler((_, parts) => OnPartitionsRevoked(ctx, parts))
                .SetPartitionsAssignedHandler((_, parts) => OnPartitionsAssigned(ctx, parts));

            var consumer = cb.Build();
            consumer.Subscribe("nosuchtopicfsdfsdfsdfs");
```

All the logs for consumer are:
```
Debug MEMBERID : [thrd:app]: Group "DefaultW": updating member id "(not-set)" -> ""
Debug INIT : [thrd:app]: librdkafka v1.2.2 (0x10202ff) d140a402ffae4c2489bb1199d8be94b3_0#consumer-2 initialized (builtin.features gzip,snappy,ssl,sasl,regex,lz4,sasl_gssapi,sasl_plain,sasl_scram,plugins,zstd,sasl_oauthbearer, SSL ZLIB SNAPPY SASL_SCRAM PLUGINS HDRHISTOGRAM, debug 0x2504)
Debug CGRPOP : [thrd:main]: Group "DefaultW" received op SUBSCRIBE (v0) in state init (join state init, v1 vs 0)
Debug SUBSCRIBE : [thrd:main]: Group "DefaultW": subscribe to new subscription of 1 topics (join state init)
Debug UNSUBSCRIBE : [thrd:main]: Group "DefaultW": unsubscribe from current unset subscription of 0 topics (leave group=no, join state init, v1)
Debug GRPLEADER : [thrd:main]: Group "DefaultW": resetting group leader info: unsubscribe
Debug REBALANCE : [thrd:main]: Group "DefaultW" is rebalancing in state init (join-state init) without assignment: unsubscribe
Debug CGRPJOINSTATE : [thrd:main]: Group "DefaultW" changed join state init -> wait-unassign (v1, state init)
Debug UNASSIGN : [thrd:main]: Group "DefaultW": unassign done in state init (join state wait-unassign): without new assignment: unassign (no previous assignment)
Debug CGRPJOINSTATE : [thrd:main]: Group "DefaultW" changed join state wait-unassign -> init (v1, state init)
Debug CGRPSTATE : [thrd:main]: Group "DefaultW" changed state init -> query-coord (v1, join-state init)
Debug CGRPQUERY : [thrd:main]: Group "DefaultW": no broker available for coordinator query: intervaled in state query-coord
```

1.2.2 Confluent.Kafka and librdkafka on Windows.
Kafka 2.3.1 I think.

Checklist
=========

Please provide the following information:
 - [x] A complete (i.e. we can run it), minimal program demonstrating the problem. No need to supply a project file.
 - [x] Confluent.Kafka nuget version.
 - [x] Apache Kafka version.
 - [x] Client configuration.
 - [x] Operating system.
 - [x] Provide logs (with "debug" : "..." as necessary in configuration).
 - [ ] Provide broker log excerpts.
 - [ ] Critical issue.

**Comment by mhowlett** [comment]
if you subscribe to a non-existant topic, I believe what happens is the consumer will automatically start reading from it as soon as it gets created. btw: I agree with you - I think it would be better if subscribe blocked on the cluster (i.e. was awaitable) and threw and exception if a topic didn't exist, then the application should determine what to do next. that seems a better arrangement to me as it seems most likely that if you subscribe to a topic, you expect it to exist.

**Comment by edenhill** [comment]
That would be a drastic change to the API, and does not really cover all corner cases (e.g., what if a topic is deleted after subscribe?), what we will do though is start propagating these type of asynchronous topic errors (no such topic, topic authorization failed, etc), through the standard consumer interface.

More info here:
https://github.com/edenhill/librdkafka/wiki/Proposal:-Consumer-topic-error-handling

**Comment by mhowlett** [comment]
From a C# perspective, it's just adding another method `SubscribeAsync`, which is not drastic - it works very naturally alongside the existing `Subscribe`. `SubscribeAsync` makes it trivial to express the behavior people most commonly want, so it's a superior API. We should do everything we can to align the API closely to typical usage patterns. Whether or not we provide `SubscribeAsync` is pretty independent of how error cases are handled after subscribe.

**Comment by mausch** [comment]
FWIW what I do as a workaround is call `IAdminClient.DescribeConfigsAsync()` before subscribing, which does throw if the topic doesn't exist.

**Comment by edenhill** [comment]
@mhowlett That means an application will need to handle the same error (topic does not exist, auth failure) in two different ways; synchronously through subscribe and asynchronously at any time after subscribe(). How is that better than just having one way?

**Comment by mhowlett** [comment]
I think you would often want different application behavior if the topic is inaccessible / doesn't exist on startup vs if the same happens during normal execution (it's the same argument as to why I think we should have blocking constructor alternative). If you don't, then no problem, you can use the same `catch` handler to cover both cases.

**Comment by flibustier7seas** [comment]
How can I detect errors (no such theme, topic authorization failed) with the .Net Kafka client, currently?

ConsumeResult doesn't have information about errors and I don't want parse debug message.

**Comment by edenhill** [comment]
This is addressed in the upcoming v1.5.0 release.
